modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/crs/NetCDFCoordinateReferenceSystemType.java

```diff
@@ -215,6 +252,32 @@ public static NetCDFCoordinateReferenceSystemType parseCRS(CoordinateReferenceSy
      */
     public abstract NetCDFProjection getNetCDFProjection();
 
+    /**
+     * Return the set of {@link NetCDFCoordinate}s for the current NetCDF CRS type, taking into
+     * account a reference crs.
+     */
+    public NetCDFCoordinate[] getCoordinates(CoordinateReferenceSystem crs) {
+        NetCDFCoordinate[] axisCoordinates = getCoordinates();
+        if (this != NetCDFCoordinateReferenceSystemType.WGS84) {
+            // NetCDF only support WGS84 and a few projections
+            // There might be the case that the projection contains a concatenatedTransform
+            // due to a specific unit in the coordinate axis (as an instance, km or foot instead
+            // of classic meter).
+            // Extract the unit of measure from the actual coordinate axis in that case.
+            ProjectedCRS projectedCrs = (ProjectedCRS) crs;
+            if (NetCDFCoordinateReferenceSystemType.hasConcatenatedTransform(projectedCrs)) {
```

Comment:
You're still in the same class, no need to qualify :-)

Reply:
Right (see above). at the beginning it was in a different place. Fixing. Thx